### PR TITLE
Replace traverse with neotraverse

### PR DESCRIPTION
**Pull request by wojtekmaj**
This PR replaces traverse with neotraverse, a fork and TypeScript rewrite of traverse with 0 dependencies (as opposed to 66: https://npmgraph.js.org/?q=traverse) and lots of improvements.

### CODE REVIEW

File: openapi-format.js
Comment (PuruVJ) [inline openapi-format.js:4]: Should be `neotraverse/legacy` 😄

Comment (wojtekmaj) [reply openapi-format.js:4]: Why? engines in package.json says Node >=18 🤔

Comment (PuruVJ) [reply openapi-format.js:4]: neotraverse doesn't ship with CommonJS. That's for /legacy build 😄

**Comment by thim81** [comment]
hi @wojtekmaj @PuruVJ 

Thanks for creating this PR, I was not aware of the dependency impact of neotraverse.
I'll review it and validate if there is any impact.

Out of curiosity, are you related (user, contributor) to "neotraverse"? or how did you end up with openapi-format?

File: openapi-format.js
Comment (thim81) [reply openapi-format.js]: The test suite is failing `Cannot find module 'neotraverse/legacy' from 'openapi-format.js'`
More details: 
https://github.com/thim81/openapi-format/actions/runs/9987096142/job/27614138151?pr=117

**Comment by PuruVJ** [comment]
Greetings @thim81!! 

I forked traverse and created the package `neotraverse` to deal with the sudden bloat in traverse(0 dep to 66!! And 58 owned directly by one maintainer). neotraverse aims to be dependency-free forever, have the same API as traverse(I monitor the repo closely for updates) and provide modern ESM-only modules. The exception is the `neotraverse/legacy` package which is only 2.43KB, compiled to ES5 and _theoretically_ compatible as far back as Node 4.0(Not intended, just a side-effect of compiling to ES5 😅).

neotraverse has not been integrated into [Builder.io Mitosis](https://x.com/puruvjdev/status/1813262143384658279), [aws-sdk-mock](https://github.com/dwyl/aws-sdk-mock/pull/443#pullrequestreview-2183271712), [feathers-hooks-common](https://github.com/feathersjs-ecosystem/feathers-hooks-common/pull/753) and discussions are happening to add it to other packages(Astro team for eg).

I would recommend using neotraverse instead of traverse because it is 0-dependency, hence your end users won't have to deal with additional bloat. You can even copy the index.ts file in the repo and maintain it yourself, which is great for security, which is not possible in `traverse` with its 66 dependencies.

And because I'm a bit biased being the creator of neotraverse, I will also recommend pinning `traverse` to 0.6.8 and never updating it if that's the direction you would prefer(traverse being longer in existence).

Hope this helps!
Regards

File: openapi-format.js
Comment (PuruVJ) [reply openapi-format.js]: I found out the issue, it's to do with how `this` in traverse methods is used and accessed. (`get(this, 'parent.parent.key')` etc). This matters because in neotraverse, internally we're using a class, whereas traverse was a function constructor with methods attached to the prototype. 

This will require refactoring some code slightly, which is up to @wojtekmaj. If he's fine with refactoring, all the better, but in case not, I'd still suggest that you pin 0.6.8 of traverse explicitly and never upgrade it(unless you're refactoring this package into ESM, in whic case neotraverse will shine 😄). That alone is enough to reduce bloat size a lot!

Comment (thim81) [reply openapi-format.js]: hi @PuruVJ,

I already appreciate that you and @wojtekmaj have taken the time to create this PR.
I'm a bit confused about the test failing because ` Cannot find module 'neotraverse/legacy' from 'openapi-format.js'` and the `this` function <> Class.

Looking at the Neotraverse docs, `this.parent.parent.key`seems to be supported, so if understand better what is wrong I could refactor it myself.

The `get(this, 'parent.parent.key')` is just little helper utility, since before NodeJS18 optional chaining was not possible, so this [get](https://github.com/thim81/openapi-format/blob/ccbd2c1d615ed51ce3c58f1a9bd6b41e0bac28c5/utils/filtering.js#L40) was used to not riddle the code with more IF statements.

Comment (PuruVJ) [reply openapi-format.js]: So, a few changes are needed:
1st: Replace `neotraverse/legacy` with `neotraverse/dist/legacy/legacy.cjs`. I am not sure why node isn't picking up the regular import, everywhere else it worked.

2nd I assume has to do with `get`, but I'd like to inspect some more.

Comment (thim81) [reply openapi-format.js]: I'll give that a try.

Comment (PuruVJ) [reply openapi-format.js]: I figured out the import issue, v0.6.12 fixes that. Could you bump it up? @wojtekmaj

**Review by thim81 — APPROVED** [review]
Looks very good. Impressive how much packages will be removed from the packages-lock.json.

**Pull request merged**
